graylog2-server/src/main/java/org/graylog/plugins/views/search/validation/FieldTypeValidationImpl.java

```diff
@@ -0,0 +1,98 @@
+
+    private static final Map<String, Predicate<String>> VALIDATION_FUNCTIONS = new HashMap<>();
+
+    private static final Predicate<String> ALWAYS_TRUE_PREDICATE = value -> true;
+
+    static {
+        VALIDATION_FUNCTIONS.put("string", ALWAYS_TRUE_PREDICATE);
+        VALIDATION_FUNCTIONS.put("long", wrapException(Long::parseLong));
+        VALIDATION_FUNCTIONS.put("int", wrapException(Integer::parseInt));
+        VALIDATION_FUNCTIONS.put("short", wrapException(Short::parseShort));
+        VALIDATION_FUNCTIONS.put("byte", wrapException(Byte::parseByte));
+        VALIDATION_FUNCTIONS.put("double", wrapException(Double::parseDouble));
+        VALIDATION_FUNCTIONS.put("float", wrapException(Float::parseFloat));
+        VALIDATION_FUNCTIONS.put("date", wrapException(DateTime::parse));
+        VALIDATION_FUNCTIONS.put("boolean", wrapException(Boolean::parseBoolean));
+        VALIDATION_FUNCTIONS.put("binary", ALWAYS_TRUE_PREDICATE);
+        VALIDATION_FUNCTIONS.put("geo-point", ALWAYS_TRUE_PREDICATE);
+        VALIDATION_FUNCTIONS.put("ip", InetAddresses::isInetAddress);
+    }
+
+    @SuppressWarnings("ReturnValueIgnored")
+    private static Predicate<String> wrapException(Function<String, Object> parser) {
+        return (value) -> {
+            try {
+                parser.apply(value);
+                return true;
+            } catch (Exception e) {
+                return false;
+            }
+        };
+    }
+
+    @Override
+    public Optional<ValidationMessage> validateFieldValueType(ParsedTerm t, String detectedFieldType) {
+        if (!typeMatching(detectedFieldType, t.value())) {
+            final ValidationMessage.Builder builder = ValidationMessage.builder(ValidationType.INVALID_VALUE_TYPE)
+                    .errorMessage(String.format(Locale.ROOT, "Type of %s is %s, cannot use value %s", t.getRealFieldName(), detectedFieldType, t.value()));
+            Stream.of(t.valueToken(), t.keyToken())
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .findFirst()
+                    .ifPresent(token -> {
+                        builder.beginLine(token.beginLine());
+                        builder.beginColumn(token.beginColumn());
+                        builder.endLine(token.endLine());
+                        builder.endColumn(token.endColumn());
+                    });
+            return Optional.of(builder.build());
+        }
+        return Optional.empty();
+    }
+
+    private boolean typeMatching(String type, String value) {
+        return Optional.ofNullable(type)
+                .map(VALIDATION_FUNCTIONS::get)
+                .map(validator -> validateValue(validator, value))
+                .orElse(true);
+    }
+
+    private Boolean validateValue(Predicate<String> validator, String value) {
+        try {
+            return validator.test(value);
+        } catch (Exception e) {
+            return false;
+        }
+    }
```

Comment:
You have wrapped exceptions in VALIDATION_FUNCTION initialization, do you have to use try-catch here in that situation?